### ISSUE TITLE
TimePicker - onChange not firing on enter, causing error to be thrown

### ISSUE THREAD

**Issue by akus062381** [patternfly/patternfly-react#8712]
Our app is using this component for a date picker. 

If you type in a time like 11:11 and then hit enter (and let the component fill in AM/PM) you'll notice in the console that onChange is never triggered and the value never gets update to include AM/PM.  This is the same thing we see in our application.  The PF component is adding AM/PM to the string but onChange  never fires so the value that we pass to the utility function is missing AM/PM. 
In short- a user editing the time is getting an error because of something having to do with AM/PM not being correctly applied.

The user that shows up in the UI looks like this:
![image](https://user-images.githubusercontent.com/63685497/219793649-3770d7cf-868c-4aeb-9467-602eedac781f.png)

**Comment by thatblindgeye** [comment]
Hey @akus062381! So when checking out our basic 12 hour Time Picker example, when I manually type in a time like "1:23" and hit Enter, the `onChange` fires and consoles the time with the "AM"/"PM".

I took a look at the linked issue in Ansible and found the DateTimePicker component (if this isn't relevant to the issue just let me know), and it looks like the Patternfly Timepicker being used there has the following onChange passed in:

```
onChange={(time) => timeHelpers.setValue(time)}
```

Last month we updated some of our beta components (including TimePicker in https://github.com/patternfly/patternfly-react/pull/8493) to add an `event` parameter to various callback props. Can you confirm whether passing an `event` before `time` in the above callback resolves the issue?

**Comment by thatblindgeye** [comment]
Closed via #8765
